field/src/extension/complex.rs

```diff
@@ -56,6 +56,14 @@ impl<AF: AbstractField> Complex<AF> {
     pub fn to_array(&self) -> [AF; 2] {
         self.value.clone()
     }
+    // Sometimes we want to rotate over an extension that's not necessarily ComplexExtendable,
+    // but still on the circle.
+    pub fn rotate<Ext: AbstractExtensionField<AF>>(&self, rhs: Complex<Ext>) -> Complex<Ext> {
```

Comment:
This function implements arbitrary complex multiplication, which agrees with `rotation` if the complex numbers being multiplied lie both on the unit circle, otherwise, it is not necessarily a rotation (there is a rescaling involved as well). I like the naming though, as it conveys the geometric transformation (on the circle). 

However, given the above observation, this naming is not strictly precise. We might want to have an assert to check that the two points lie in the circle (`z * \conjugate{z} = 1`), or then rename it to `complex_multiplication`.

I also suggest to had the above observations in the documents, in some form.

Reply:
I will add a `debug_assert` that both points are on the circle. I'm not a mathematician, but I'm not sure if `complex_multiplication` is correct, as the intent is we can use this method even if the extension does not have a complex extension (`p mod 4 != 3`). Actually what's happening is that the code often uses `Complex` as a shorthand for affine (X,Y) coordinates, and it's usually the same thing, but slightly inaccurate when moving to the extension for the DEEP challenge

Reply:
The most accurate name would be something like `circle_group_operation`... the paper calls it $T$